File: 05_do_while.js
// 05_do_while.js

var random_number = Math.random() * 100 + 1;
var int_random_number = Math.floor(random_number);
var counter = 0;

document.write("Random number: ", int_random_number, "<br>");
document.write("<br>");
document.write("Iterations until random number (i=5).<br>");
document.write("Last iteration is a smaller step (i<5).<br>");
document.write("<br>Iterations:<br>");
do 
{
    if (counter < int_random_number) {
        counter++;
        if (counter%5 === 0) {
            document.write(counter, ", ");
        }
    } 
}
while (int_random_number != counter);

if (int_random_number === counter) {
    document.write(counter, "<br>");
}

File: 05_do_while.py
#!/usr/bin/env python3

# Python 3.9.5

# 05_do_while.py

# Please note: A do-while-loop does not exist in Python3.
# A workaround is the regular while loop.

# Dependency
import random

random_number = random.randint(1, 100)
counter = 0
counter_lst = []

print('Random number:', random_number)
print('Iterations until random number (i=5)')
print('Last iteration is a smaller step (i<5)')
print('Iterations:')

while counter < random_number:
    counter += 1
    if (counter % 5) == 0:
        counter_lst.append(counter)

if counter == random_number: 
    counter_lst.append(counter)

print(counter_lst)
